packages/replay/src/replay.ts

```diff
@@ -253,14 +259,28 @@ export class ReplayContainer implements ReplayContainerInterface {
         log(msg);
       }
 
+      // Set this property so that it ignores `_isEnabled` = false
+      // We can't move `_isEnabled` after awaiting a flush, otherwise we can
+      // enter into an infinite loop when `stop()` is called while flushing.
+      this._shouldFinalFlush = true;
```

Comment:
Hmm, I don't really like this flag - I would prefer we find a way to solve this by e.g. passing an argument to `flushImmediate()` or making a separate function for this type of flushing to ensure this. Seems prone to run out of sync, be accidentally true/false when we don't mean it to, etc. 

One option I could see is to just do:

```js
this._isEnabled = false;
this._removeListeners();
this._flushAfterStop();

function _flushAfterStop() {
  this._debouncedFlush().cancel();
  this._flush({ force: true });
}
```

Or something along these lines? IMHO as we are stopping here anyhow, we don't necessarily need to run the debounced flush here at all, allowing us to separate this a bit better?

Reply:
updated, instead of creating a private method, I just kept the flush calls inline as it isn't used elsewhere.